### ISSUE TITLE
ClearDragDrop() invoking only on EndFrame()

### ISSUE THREAD

**Issue by DimaKoltun** [ocornut/imgui#5817]
----
**Version/Branch of Dear ImGui:**

Version: 1.89 WIP
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_sdl.cpp + imgui_impl_opengl3.cpp
Compiler: MSVC
Operating System: Windows 10

**My Issue/Question:**

I want to give keyboard focus to input text in modal dialog that appearing on drop something to window, but input text doesn't get focus due to DragDropActive is true; DragDropActive set to false only on invoke EndFrame() -> ClearDragDrop(), may be ClearDragDrop() should be invoked earlier, for example, in AcceptDragDropPayload() if it's return true?

**Screenshots/Video**

https://user-images.githubusercontent.com/83829508/197537294-3a298cd8-279e-4caf-924e-be519979b280.mp4

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```
        // Here's some code anyone can copy and paste to reproduce your issue
        char buf[64] = {};        

        //-------
        ImGui::Begin("Window");
        if (ImGui::Button("Drag this"))
        {

        }
        int dndData = 0;
        if (ImGui::BeginDragDropSource(ImGuiDragDropFlags_None))
        {
            ImGui::SetDragDropPayload("DND_TEST", &dndData, sizeof(int));
            ImGui::EndDragDropSource();
        }

        if (ImGui::Button("Drop Here"))
        {
        }
        if (ImGui::BeginDragDropTarget())
        {
            if (const ImGuiPayload* payload = ImGui::AcceptDragDropPayload("DND_TEST"))
            {
                ImGui::OpenPopup("Modal");
            }

            ImGui::EndDragDropTarget();
        }

        if (ImGui::Button("Click to open modal dialog"))
        {
            ImGui::OpenPopup("Modal");
        }

        if (ImGui::BeginPopupModal("Modal"))
        {
            if (ImGui::GetCurrentWindow()->Appearing)
            {
                ImGui::SetKeyboardFocusHere();
            }

            ImGui::InputText("Name", buf, 64);

            if (ImGui::Button("Close"))
            {
                ImGui::CloseCurrentPopup();
            }

            ImGui::EndPopup();
        }

        ImGui::End();
```

**Comment by ocornut** [comment]
Thank you for the detailed report and repro, much appreciated.

Suggested change seems reasonable (we could perform the clear `EndDragDropTarget()`)I will need to think of potential side-effects. I'm reviewing all DnD tasks and issues right now.

**Comment by ocornut** [comment]
For now I suggest you call `ClearDragDrop()` yourself _after_ `EndDragDropTarget()`.

**Comment by ocornut** [comment]
I have applied two fixes as part of e9743d8
And pushed tests in https://github.com/ocornut/imgui_test_engine/commit/b5d1ecc8ac2ef4c1c24a8e74775e5e9cfdad8142
Thank you!
See also #6183
